lib/cfn-nag/custom_rule_loader.rb

```diff
@@ -193,12 +195,17 @@ def discover_rule_classes(rule_directory)
 
     rule_filenames.each do |rule_filename|
       require(rule_filename)
-      rule_classname = File.basename(rule_filename, '.rb')
+    end
 
-      rule_classes << Object.const_get(rule_classname)
+    ObjectSpace.each_object do |object|
```

Comment:
i'm not usually one to care *so* much about performance but.... this iterates over every object in the heap?  what kind of performance price do we pay for this? (or can we pass BaseRule to filter?)

Reply:
is this looking for Class objects somehow v. any/every object instance?

Reply:
This is looking at all objects in the heap. I'll look into ways to increase performance. I think rule detection may be happening multiple times so one optimization may be to cache the rules found the first time and keep using them.

Reply:
@erickascic performance has been addressed.